src/instrumentation/page.ts

```diff
@@ -0,0 +1,85 @@
+import { ReadableSpan } from '@opentelemetry/sdk-trace-base'
+import { Initialiser, setConfig } from '../config'
+import { exportSpans, proxyExecutionContext } from './common'
+import { Exception, SpanKind, SpanOptions, SpanStatusCode, context as api_context, trace } from '@opentelemetry/api'
+import { wrap } from '../wrap'
+import {
+	gatherIncomingCfAttributes,
+	gatherRequestAttributes,
+	gatherResponseAttributes,
+	getParentContextFromRequest,
+} from './fetch'
+
+type PageHandlerArgs = Parameters<PagesFunction>
+
+let cold_start = true
```

Comment:
We should make this a bit longer to help ensure it doesn't ever conflict with user code (generally, global vars should be avoided in Workers/Functions, but this seems ok.)

e.g. `let __otel_cf_is_cold_start = true`

Reply:
I think this should actually be fine, as it's not exported, so user code can never find it as `cold_start`.